File: redsheet-portal/src/app/database-models/v-stakeholder-communication-mode.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vStakeholderCommunicationMode {
    constructor(vStakeholderCommunicationMode?: vStakeholderCommunicationMode )  {
        let properties = [ "communicationModeId", "communicationModeName", "projectStakeholderId", "stakeholderCommunicationModeId",];
        for (let property of properties)
            if (vStakeholderCommunicationMode && vStakeholderCommunicationMode[property])
                this[property] = vStakeholderCommunicationMode[property];
    }
 
	communicationModeId : number =   0 ;
 
    @required()
    @maxLength(200)
	communicationModeName : string =   undefined;
 
    @range(1,2147483647)
	projectStakeholderId : number =   undefined;
 
    @range(1,2147483647)
	stakeholderCommunicationModeId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vStakeholderCommunicationMode.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vStakeholderCommunicationModes", Schema = "dbo")]
	[Serializable]
	public partial class vStakeholderCommunicationMode  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CommunicationModeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string CommunicationModeName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectStakeholderId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int StakeholderCommunicationModeId { get; set; } 

	public vStakeholderCommunicationMode()
	{
    }

    }
}

